src/config.ts

```diff
@@ -55,23 +59,25 @@ export const updateCurrentConfig = (siteCfg, _directives) => {
  * @param conf - The base currentConfig to use as siteConfig
  * @returns {object} - The siteConfig
  */
-export const setSiteConfig = (conf) => {
+export const setSiteConfig = (conf: MermaidConfig): MermaidConfig => {
   siteConfig = assignWithDepth({}, defaultConfig);
   siteConfig = assignWithDepth(siteConfig, conf);
 
+  // @ts-ignore
   if (conf.theme && theme[conf.theme]) {
+    // @ts-ignore
     siteConfig.themeVariables = theme[conf.theme].getThemeVariables(conf.themeVariables);
```

Comment:
```suggestion
  if (conf.theme && conf.theme in theme) {
    siteConfig.themeVariables = theme[conf.theme as keyof typeof theme].getThemeVariables(conf.themeVariables);
```

We can remove the `@ts-ignore`s here too by using `in` and making an assertion.